### PR TITLE
DAOS-5065 csum: Checksum Fixes for CI Step Unit Test

### PR DESCRIPTION
**Pull request by ryon-jensen**
Resolved a few issues discovered in CI when checksums
are enabled by default. The primary failure was run_test.sh
wasn't passing for a few reasons:

- cart log test had several false positives on mismatched alloc/free.
- an unnecessary warning was logged when a request was larger than data
  in sgl
- memory not freed for containers opened as a copy of an existing open
  container. (When opened from g2l).
- some checksum tests weren't cleanup up after themselves.

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
> Resolved a few issues discovered in CI when checksums
> are enabled by default.

This PR all looks sensible so what's the cause of these issues?  Have we changed something else in a different PR that introduced regressions that we're now fixing in this one?  I'd have thought we'd typically fix the issues in the same PR that enabled the feature, or it not at least link to it here.

**Comment by ryon-jensen** [comment]
> > Resolved a few issues discovered in CI when checksums
> > are enabled by default.
> 
> This PR all looks sensible so what's the cause of these issues? Have we changed something else in a different PR that introduced regressions that we're now fixing in this one? I'd have thought we'd typically fix the issues in the same PR that enabled the feature, or it not at least link to it here.

That's a good point. @rpadma2 created a PR that enables the checksum feature by default in order to discover checksum issues in CI.  Currently, that PR isn't intended to be landed but to just help build confidence in the checksum feature. There is still some debate on whether or not to enable checksums by default.  Link to that PR: https://github.com/daos-stack/daos/pull/2796

**Comment by ryon-jensen** [comment]
Added another small change to fix an issue to a unit test that @wangdi1 discovered.

File: src/cart/test/util/cart_logtest.py [hunk 1]
```diff
@@ -117,10 +117,13 @@ class LogError(LogCheckError):
                      'mgmt_svc_name_cb': ('s'),
                      'pool_prop_default_copy': ('entry_def->dpe_str'),
                      'pool_iv_prop_g2l': ('prop_entry->dpe_str'),
-                     'pool_iv_value_alloc_internal': ('sgl->sg_iovs[0].iov_buf'),
+                     'pool_iv_value_alloc_internal':
+                             ('sgl->sg_iovs[0].iov_buf'),
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/cart/test/util/cart_logtest.py:121]: These entries are never going to fit within the 80 character limit and continuations like this just make the code block longer so I've just been doing them all one-per-line.

[the remark this answers is not in the document]
Comment (ryon-jensen) [reply src/cart/test/util/cart_logtest.py:121]: Will revert

**Comment by ryon-jensen** [comment]
> We should have some tests that run in this mode, even if it's not the default. The node local testing seemed to find a couple of issues so perhaps enable it there for now?

There is a suite of tests with checksum enabled in daos_test, at the object layer, in vos, and in common. As we've found bugs through enabling across all of CI, I've tried to add a test in the appropriate layer to make sure the bug doesn't happen again. These failures are more difficult to do that for because they are memory related.

**Comment by ryon-jensen** [comment]
IOR Small Test failure: DAOS-4989
Rebuild 26 test: DAOS-5498

**Pull request merged**